feedstock/recipe.py

```diff
@@ -1,43 +1,70 @@
-import pandas as pd
-from datetime import datetime, date
+from datetime import datetime, date, timedelta
+from os.path import basename
 from pangeo_forge_recipes.patterns import ConcatDim, MergeDim, FilePattern
 from pangeo_forge_recipes.recipes import XarrayZarrRecipe
 
-# RIOPS is run every 6 hours (at 00, 06, 12, and 18 UTC).  
-# but, for the moment, let's use today's date and assume that when this recipe is being run,
-# the 00UTC run for today is already available!
-start_date = date.today()
-time00 = datetime.min.time()
-start_time = datetime.combine(start_date, time00)
-
+# RIOPS is run every 6 hours (at 00, 06, 12, and 18 UTC).
+# but, MSC usually only has 2-days-worth of data on the datamart.
+# Thus, we use yesterday's date to ensure the feedstock can run
 def make_url(variable, time):
-
-    yyyymmdd = start_date.strftime("%Y%m%d")
-    index = f'{time:03.0f}'
-    return (
+    """Currently hardcoded for 2D variables run at 00Z"""
+    riops_ps_base_url = (
         "https://dd.weather.gc.ca/model_riops/netcdf/forecast/polar_stereographic"
-        f"/2d/00/{index}/{yyyymmdd}T00Z_MSC_RIOPS_{variable}_DBS-0.5m_PS5km_P{index}.nc"
     )
+    start_date = datetime.combine(date.today() - timedelta(days=1), datetime.min.time())
```

Comment:
let's move the `datetime` imports inside the function bodies as well. 

the latest run failed with this error

```python
  File "/srv/conda/envs/notebook/lib/python3.9/site-packages/pangeo_forge_recipes/recipes/xarray_zarr.py", line 155, in cache_input
    fname = config.file_pattern[input_key]
  File "/srv/conda/envs/notebook/lib/python3.9/site-packages/pangeo_forge_recipes/patterns.py", line 219, in __getitem__
    fname = self.format_function(**format_function_kwargs)
  File "/tmp/tmpih5qufhy/feedstock/recipe.py", line 14, in make_url
NameError: name 'datetime' is not defined [while running 'Start|cache_input|Reshuffle_000|prepare_target|Reshuffle_001|store_chunk|Reshuffle_002|finalize_target|Reshuffle_003/cache_input/Execute-ptransform-56']
```

Reply:
I will also move the `from os.path import basename` inside the function where it's used (just-in-case)

Reply:
still happily builds and loads locally as of https://github.com/pangeo-forge/riops-feedstock/pull/4/commits/fd14354891660b7992f48a6a6f3f7c596bb0e894